### PR TITLE
Import libgit4cpp from taskolib

### PR DESCRIPTION
**Pull request by alt-graph**
This PR imports the git functionality from Taskolib and makes a stand-alone library out of it. A few modifications of the original source code were necessary to get it into shape. Please have a look.

### CODE REVIEW

**Comment by alt-graph** [comment]
@wyrnat: Oh, I forgot: I have renamed the repository from gitlib4cpp to libgit4cpp. The original project is called "libgit2" and the Unix library name would otherwise become "libgitlib4cpp"...

**Comment by ohensler** [comment]
Nice stuff

**Review by ohensler — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -0,0 +1,94 @@
+libgit4cpp
+==========
+
+libgit4cpp is a lightweight C++ wrapper around the [libgit2](https://libgit2.org/) C
+library. It was mainly developed for use in
+[taskolib](https://taskolib.github.io/taskolib/) and covers only a tiny subset of the
+functionality of the original library.
+
+The library API is documented with Doxygen on the web page
+https://taskolib.github.io/libgit4cpp/ .
```
Comment (wyrnat) [inline README.md:10]: The page does not exist

**Review by wyrnat — APPROVED** [review]
I have nothing to add. Looks good and I like the decision to separate wrapper functions and wrapper types in two files

File: README.md (hunk 1 above)
Comment (alt-graph) [reply README.md:10]: I think it will come into existence once the Github actions in this PR are active. I'll merge and we'll have a look.

**Pull request merged**